lstchain/scripts/longterm_dl1_check.py

```diff
@@ -170,18 +221,39 @@ class pixwise_info(tables.IsDescription):
             try:
                 node = a.get_node(name)
             except Exception:
-                print('   Table', name, 'is missing!')
+                log.warning('Table {name} is missing!')
                 datatables.append(None)
                 continue
 
             datatables.append(node)
 
         subruns = None
 
+        trig_tags = [TriggerBits.PHYSICS.value,
+                     TriggerBits.PEDESTAL.value,
+                     TriggerBits.CALIBRATION.value]
+
         # fill data which are common to all tables:
-        for table, d in zip(datatables, dicts):
+
+        total_num_ucts_jumps = 0  # To add up all jumps, for any event type
+
+        for table, d, tag in zip(datatables, dicts, trig_tags):
             if table is None:
                 continue
+            # check the number of wrong trigger types, i.e. those events which
+            # do not have the expected trigger bits for a cosmic, pedestal or
+            # flatfield event, depending on the table we are processing. Note
+            # that the events were classified in the tables (see
+            # dl1_checker.py) using event_type as filled by the LST event source
+            #
+            # num_wrong_tags = [ndarray, ndarray]  (for ucts & tib respectively)
+            # Each array has one entry per subrun, with the number of mismatches
+            num_wrong_tags = trigtag_mismatches(table, tag)
+            d['wrong_ucts_trig_type'].extend(num_wrong_tags[0])
+            d['wrong_tib_trig_type'].extend(num_wrong_tags[1])
+            d['num_ucts_jumps'].extend(table.col('num_ucts_jumps'))
```

Comment:
`num_ucts_jumps` key is not in the data files we have already produced, therefore this will fail with those already processed data. I guess it will work for newly processed data.

```python
Traceback (most recent call last):
  File "/fefs/aswg/workspace/daniel.morcuende/.conda/envs/lstchain-0.7.6dev/lib/python3.8/site-packages/tables/table.py", line 1150, in _get_column_instance
    return _reduce(getattr, colpathname.split('/'), self.description)
AttributeError: 'Description' object has no attribute 'num_ucts_jumps'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "lstchain/scripts/longterm_dl1_check.py", line 1340, in <module>
    main()
  File "lstchain/scripts/longterm_dl1_check.py", line 254, in main
    d['num_ucts_jumps'].extend(table.col('num_ucts_jumps'))
  File "/fefs/aswg/workspace/daniel.morcuende/.conda/envs/lstchain-0.7.6dev/lib/python3.8/site-packages/tables/table.py", line 2023, in col
    return self.read(field=name)
  File "/fefs/aswg/workspace/daniel.morcuende/.conda/envs/lstchain-0.7.6dev/lib/python3.8/site-packages/tables/table.py", line 1919, in read
    self._check_column(field)
  File "/fefs/aswg/workspace/daniel.morcuende/.conda/envs/lstchain-0.7.6dev/lib/python3.8/site-packages/tables/table.py", line 1152, in _get_column_instance
    raise KeyError("table ``%s`` does not have a column named ``%s``"
KeyError: 'table ``/dl1datacheck/cosmics`` does not have a column named ``num_ucts_jumps``'
```
We could catch this exception in case jump info is not present so the script can keep going.

Reply:
It is in principle not intended to be re-run on old data - I tried on a subrun newly produced and it works.
But you are right, it might be needed - I will simply skip it (leave it to the default -1) in case ucts_jump is not there.

Reply:
Should work now with both old and new DL1*h5 files and datacheck*h5 files